packages/console/src/pages/SignInExperience/utilities.ts

```diff
@@ -0,0 +1,55 @@
+import { SignInExperience, SignInMethods, SignInMethodState } from '@logto/schemas';
+
+import { SignInExperienceForm } from './types';
+
+const findMethodState = (
+  setup: SignInExperienceForm,
+  method: keyof SignInMethods
+): SignInMethodState => {
+  const { signInMethods } = setup;
+
+  if (signInMethods.primary === method) {
+    return SignInMethodState.primary;
+  }
+
+  if (signInMethods[method]) {
+    return SignInMethodState.secondary;
+  }
+
+  return SignInMethodState.disabled;
+};
+
+export const signInExperienceParser = {
+  toLocalForm: (signInExperience: SignInExperience): SignInExperienceForm => {
+    const methodKeys: Array<keyof SignInMethods> = ['username', 'email', 'sms', 'social'];
```

Comment:
nit: better extract `SignInMethodKey`?
<img width="731" alt="image" src="https://user-images.githubusercontent.com/10594507/163525594-7207aed7-cf6d-4cb8-9356-fd560366fe45.png">

and then you can use
`const methodKeys = Object.values(SignInMethodKey);`

Reply:
@wangsijie PTAL #556

Reply:
updated